bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/transactions/InMemoryRecoveryLog.java

```diff
@@ -0,0 +1,95 @@
+/*
+ *  Copyright (c) 2024, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.runtime.transactions;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import static io.ballerina.runtime.transactions.TransactionConstants.IN_MEMORY_CHECKPOINT_INTERVAL;
+
+public class InMemoryRecoveryLog implements RecoveryLog {
+     private static final Logger log = LoggerFactory.getLogger(InMemoryRecoveryLog.class);
```

Comment:
Is the indentation right in this line?

Reply:
no, it was not. formatting mistakes were fixed in commit [5c63a68](https://github.com/ballerina-platform/ballerina-lang/pull/42080/commits/5c63a684e7c8c29d6b59d8efc333a9ed01e14e70)